relay-dynamic-config/src/metrics.rs

```diff
@@ -124,3 +125,178 @@ impl TransactionMetricsConfig {
+    pub category: DataCategory,
+
+    /// Identifier of the metric to extract.
+    pub mri: String,
+
+    /// A path to the field to extract the metric from.
+    ///
+    /// This value contains a fully qualified expression pointing at the data field in the payload
+    /// to extract the metric from. It follows the
+    /// [`FieldValueProvider`](relay_sampling::FieldValueProvider) syntax that is also used for
+    /// dynamic sampling.
+    ///
+    /// How the value is treated depends on the metric type:
+    ///
+    /// - **Counter** metrics are a special case, since the default product counters do not count
+    ///   any specific field but rather the occurrence of the event. As such, there is no value
+    ///   expression, and the field is set to `None`. Semantics of specifying remain undefined at
+    ///   this point.
+    /// - **Distribution** metrics require a numeric value. If the value at the specified path is
+    ///   not numeric, metric extraction will be skipped.
+    /// - **Set** metrics are require a string value, which is then emitted into the set as unique
+    ///   value. Insertion of numbers and other types is undefined.
+    ///
+    /// If the field does not exist, extraction is skipped.
+    #[serde(default)]
+    pub field: Option<String>,
+
+    /// An optional condition to apply before extraction.
+    ///
+    /// See [`RuleCondition`] for all available options to specify and combine conditions. If no
+    /// condition is specified, the metric is extracted unconditionally.
+    #[serde(default)]
+    pub condition: Option<RuleCondition>,
+
+    /// A list of tags to add to the metric.
+    ///
+    /// Tags can be conditional, see [`TagSpec`] for configuration options. For this reason, it is
+    /// possible to list tag keys multiple times, each with different conditions. The first matching
+    /// condition will be applied.
+    #[serde(default)]
+    pub tags: Vec<TagSpec>,
+}
+
+/// Mapping between extracted metrics and additional tags to extract.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TagMapping {
+    /// A list of metrics identifiers (MRI) to apply tags to.
+    ///
+    /// Entries in this list can contain wildcards to match metrics with dynamic MRIs.
+    #[serde(default)]
+    pub metrics: Vec<String>,
+
+    /// A list of tags to add to the metric.
+    ///
+    /// Tags can be conditional, see [`TagSpec`] for configuration options. For this reason, it is
+    /// possible to list tag keys multiple times, each with different conditions. The first matching
+    /// condition will be applied.
+    #[serde(default)]
+    pub tags: Vec<TagSpec>,
+}
+
+/// Configuration for a tag to add to a metric.
+///
+/// Tags values can be static if defined through `value` or dynamically queried from the payload if
+/// defined through `field`. These two options are mutually exclusive, behavior is undefined if both
+/// are specified.
```

Comment:
So this is the false statement here, if both specified we always return `field` as a `source` on line 255.
Or will be there more code which will make this behavior undefined ?

Reply:
My intent was to leave behavior undefined on the protocol level (hence the doc comment), but of course we need to do *something*. At this point, I'd like to leave our options open to throw an error, use both as a fallback, or just have one take precedence over the other. Either way, at this point we do not expect Sentry to emit entries with both set.